web/src/engine/keyboard/src/kmxPlus/ldmlKeyboardObject.ts

```diff
@@ -0,0 +1,77 @@
+import { decodeNumber, decodeSectName } from './utils.js';
+
+import { Keys } from './keys.js';
+import { Layr } from './layr.js';
+import { List } from './list.js';
+import { Sect } from './sect.js';
+import { Strs } from './strs.js';
+
+class UnusedSection {
+  readonly name: string;
+  readonly data: Uint8Array;;
+
+  constructor(rawData: Uint8Array, name: string) {
+    this.data = rawData;
+    this.name = name;
+  }
+}
+
+type Section = Keys | Layr | List | Sect | Strs | UnusedSection;
+
+export class LdmlKeyboardObject {
+  readonly dataBuffer: Uint8Array;
+
+  readonly keys: Keys;
+  readonly layr: Layr;
+  readonly list: List;
+  readonly sect: Sect;
+  readonly strs: Strs;
+
+  constructor(rawData: Uint8Array, startIndex: number) {
+    let kmxPlusLen = decodeNumber(rawData, startIndex + 8);
+    this.dataBuffer = new Uint8Array(rawData.buffer, rawData.byteOffset + startIndex, kmxPlusLen);
+    this.sect = this.getSection(this.dataBuffer, 0) as Sect;
+    if(this.sect.name != 'sect') {
+      throw new Error("Provided KMX+ data did not start with the `sect` table!")
+    }
+
+    /*
+     * Preferred order:
+     * - strs
+     * - list - needs strs
+     * - keys - needs lsts
+     * - layr - needs keys, others.
```

Comment:
`disp` could probably go after `strs` or `list` - it only really needs to reference `strs`, I believe.

Reply:
correct